### PR TITLE
Launch daemon from conda directory if conda is enabled

### PR DESCRIPTION
**Pull request by gmanis**
Currently, executors do not use the correct path to launch the R daemon/worker when using conda. This PR updates rLibDir to use the conda library path if conda is being used.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/api/r/RRunner.scala [hunk 1]
```diff
@@ -350,8 +350,12 @@ private[r] object RRunner {
     val rConnectionTimeout = sparkConf.getInt(
       "spark.r.backendConnectionTimeout", SparkRDefaults.DEFAULT_CONNECTION_TIMEOUT)
     val rOptions = "--vanilla"
-    val rLibDir = RUtils.sparkRPackagePath(isDriver = false)
-    val rExecScript = rLibDir(0) + "/SparkR/worker/" + script
+    val rLibDir = condaEnv.map { conda =>
+       RUtils.sparkRPackagePath(isDriver = false) :+ (conda.condaEnvDir + "/lib/R/library")
+    }.getOrElse(RUtils.sparkRPackagePath(isDriver = false))
+    print(rLibDir + "\n")
```
Comment (robert3005) [inline core/src/main/scala/org/apache/spark/api/r/RRunner.scala]: remove the prints

File: core/src/main/scala/org/apache/spark/api/r/RUtils.scala [hunk 2]
```diff
@@ -95,6 +95,12 @@ private[spark] object RUtils {
     }
   }
 
+  /** Finds the rLibDir with SparkR installed on it. */
+  def sparkRInstallLocation(rLibDir: Seq[String], scriptPath: String): String = {
+    rLibDir.find( dir => new File(dir + scriptPath).exists)
```
Comment (robert3005) [inline core/src/main/scala/org/apache/spark/api/r/RUtils.scala]: unnecessary whitespace after find.

Isn't there nicer way to check if file exists?

Comment (gmanis) [reply core/src/main/scala/org/apache/spark/api/r/RUtils.scala]: If you prefer, I can do it with something like `Files.exists(Paths.get(dir + scriptPath))` but the rest of the file uses the above pattern, so I figured I would match.

File: core/src/main/scala/org/apache/spark/api/r/RRunner.scala [hunk 3]
```diff
@@ -350,8 +350,10 @@ private[r] object RRunner {
     val rConnectionTimeout = sparkConf.getInt(
       "spark.r.backendConnectionTimeout", SparkRDefaults.DEFAULT_CONNECTION_TIMEOUT)
     val rOptions = "--vanilla"
-    val rLibDir = RUtils.sparkRPackagePath(isDriver = false)
-    val rExecScript = rLibDir(0) + "/SparkR/worker/" + script
+    val rLibDir = condaEnv.map { conda =>
+       RUtils.sparkRPackagePath(isDriver = false) :+ (conda.condaEnvDir + "/lib/R/library")
```
Comment (dansanduleac) [inline core/src/main/scala/org/apache/spark/api/r/RRunner.scala]: why do we append the condaEnvDir-based path rather than having only that one?

File: circle.yml [hunk 4]
```diff
@@ -12,8 +12,8 @@ machine:
     - |
       if [[ ! -d ${CONDA_ROOT} ]]; then
           echo "Installing Miniconda...";
-          wget --quiet https://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh &&
-            bash Miniconda-latest-Linux-x86_64.sh -b -p ${CONDA_ROOT};
+          wget --quiet https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh &&
```
Comment (dansanduleac) [inline circle.yml]: do we need to change this?

File: core/src/main/scala/org/apache/spark/api/r/RRunner.scala (hunk 3 above)
Comment (gmanis) [reply core/src/main/scala/org/apache/spark/api/r/RRunner.scala]: It's possible to have the cluster install sparkr.zip onto the executors even while using conda. In particular, the tests do not install SparkR and PySpark via conda.

File: circle.yml (hunk 4 above)
Comment (gmanis) [reply circle.yml]: Yes– the change does not specifically relate to what this PR was originally intended for, but at some point in the last couple of days the old version of conda that we were pulling in started to run into a circular dependency issue with the packages we install for the R test suite.
